File: Gemini_Flash/Java/Factory_Method/round3/source/CoffeeShop.java
package Gemini_Flash.Java.Factory_Method.round3.source;

class CoffeeShop extends CoffeeFactory {
    @Override
    Coffee createCoffee(String type) {
        switch (type.toLowerCase()) {
            case "americano":
                return new Americano();
            case "latte":
                return new Latte();
            case "cappuccino":
                return new Cappuccino();
            case "espresso":
                return new Espresso();
            case "mocha":
                return new Mocha();
            default:
                System.out.println("Coffee type " + type + " not found on the menu");
                return null;
        }
    }
}

File: Gemini_Flash/Python/Factory_Method/round3/source/coffee_shop.py
from abc import ABC, abstractmethod

class Coffee(ABC):
    @abstractmethod
    def prepare(self):
        pass

    @abstractmethod
    def serve(self):
        pass

class Americano(Coffee):
    def prepare(self):
        print("Brewing strong coffee...")
        print("Adding hot water...")

    def serve(self):
        print("Serving Americano with a glass of water.")

class Latte(Coffee):
    def prepare(self):
        print("Brewing espresso...")
        print("Adding steamed milk...")

    def serve(self):
        print("Serving Latte with latte art.")

class Cappuccino(Coffee):
    def prepare(self):
        print("Brewing espresso...")
        print("Adding steamed milk...")
        print("Adding foamed milk...")

    def serve(self):
        print("Serving Cappuccino with latte art.")

class Espresso(Coffee):
    def prepare(self):
        print("Brewing espresso...")

    def serve(self):
        print("Serving Espresso in a small cup.")

class Mocha(Coffee):
    def prepare(self):
        print("Brewing espresso...")
        print("Adding chocolate syrup...")
        print("Adding steamed milk...")
        print("Adding whipped cream...")

    def serve(self):
        print("Serving Mocha with a sprinkle of chocolate powder.")

class CoffeeFactory:
    @staticmethod
    def create_coffee(coffee_type):
        if coffee_type == "Americano":
            return Americano()
        elif coffee_type == "Latte":
            return Latte()
        elif coffee_type == "Cappuccino":
            return Cappuccino()
        elif coffee_type == "Espresso":
            return Espresso()
        elif coffee_type == "Mocha":
            return Mocha()
        else:
            raise ValueError(f"Invalid coffee type: {coffee_type}")
